Gems/Multiplayer/Code/Include/Multiplayer/IMultiplayer.h

```diff
@@ -163,6 +164,11 @@ namespace Multiplayer
         //! @param handler The NoServerLevelLoadedEvent handler to add
         virtual void AddNoServerLevelLoadedHandler(NoServerLevelLoadedEvent::Handler& handler) = 0;
 
+        //! Adds a VersionMismatchEvent Handler which is invoked whenever a connection is made between 2 apps that have differing multiplayer components.
+        //! Provided for catching an out-of-date server or client.
+        //! @param handler The VersionMismatchEvent handler to add
+        virtual void AddVersionMismatchEvent(VersionMismatchEvent::Handler& handler) = 0;
```

Comment:
Are any of these events exposed to scripting?

Reply:
Negative! I'll make that a separate PR since this PR will help the original issue found in testing which was 2 people trying to connect but they were using different builds and didn't know it. This PR shows onscreen viewport debug message when a version mismatch occurs so they wont miss it next time.
https://github.com/o3de/o3de/issues/13269